probe-rs/src/debug/unit_info.rs

```diff
@@ -68,48 +68,43 @@ impl UnitInfo {
         let mut entries_cursor = self.unit.entries();
         while let Ok(Some((_depth, current))) = entries_cursor.next_dfs() {
             let Some(mut die) = FunctionDie::new(current.clone(), self) else {
+                // We only want to process DIEs that are functions.
                 continue;
             };
 
-            let mut ranges = debug_info.dwarf.die_ranges(&self.unit, current)?;
-
-            while let Ok(Some(range)) = ranges.next() {
-                if !range.contains(address) {
-                    continue;
-                }
-
-                // Check if we are actually in an inlined function
-                die.low_pc = range.begin;
-                die.high_pc = range.end;
+            let mut gimli_ranges = debug_info.dwarf.die_ranges(&self.unit, current)?;
+            while let Ok(Some(gimli_range)) = gimli_ranges.next() {
+                die.ranges.push(gimli_range.begin..gimli_range.end);
```

Comment:
I don't see what we gain here by transforming gimli Ranges here

Reply:
A couple of reasons. The `gimli::Range` doesn't have the same methods, and uses begin/end vs start/end. I don't want to propogate the inconsistencies and in the breakpoint work I'm doing, I need the `is_empty` and `contains` methods.